CMakeLists.txt

```diff
@@ -251,84 +193,100 @@ if (NOT EXISTS ${PROJECT_SOURCE_DIR}/lib/zlib/CMakeLists.txt)
-add_subdirectory(lib/tolua++/ EXCLUDE_FROM_ALL)
-add_subdirectory(lib/SQLiteCpp/)
-add_subdirectory(lib/sqlite/)
-add_subdirectory(lib/expat/)
-add_subdirectory(lib/luaexpat/)
-add_subdirectory(lib/libevent/ EXCLUDE_FROM_ALL)
-add_subdirectory(lib/fmt)
+# Enable PCH and jumbo builds on supporting CMake:
+if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.16")
+	if (PRECOMPILE_HEADERS)
+		target_precompile_headers(${CMAKE_PROJECT_NAME} PRIVATE src/Globals.h)
+	endif()
 
+	if (UNITY_BUILDS)
+		set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES UNITY_BUILD ON)
+	endif()
+else()
+	message(WARNING "Precompiled headers for FASTER BUILDS not enabled, upgrade to CMake 1.16 or newer!")
+endif()
 
-# Add proper includes for LibEvent's event-config.h header:
-include_directories(SYSTEM ${LIBEVENT_INCLUDE_DIRS})
+# Add required includes:
+target_include_directories(
+	${CMAKE_PROJECT_NAME} SYSTEM PRIVATE
+	lib/mbedtls/include
+	lib/TCLAP/include
+	lib # TODO fix files including zlib/x instead of x
+)
+
+# Link dependencies as private:
+target_link_libraries(
+	${CMAKE_PROJECT_NAME} PRIVATE
+	event_core
+	event_extra
+	fmt::fmt
+	jsoncpp_lib
+	lsqlite
+	lua
+	luaexpat
+	mbedtls
+	SQLiteCpp
+	tolualib
+	zlib
+)
+
+# Link process information library:
+if (WIN32)
+	target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE Psapi.lib)
+endif()
 
-# Prettify jsoncpp_lib name in VS solution explorer
+# Special case handling for libevent pthreads:
+if(NOT WIN32)
+	target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE event_pthreads_static)
+endif()
+
+# Prettify jsoncpp_lib name in VS solution explorer:
 set_property(TARGET jsoncpp_lib PROPERTY PROJECT_LABEL "jsoncpp")
-# jsoncpp uses these for ccache support, clashing with our method
-set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "")
-set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK "")
 
 if (WIN32)
-	add_subdirectory(lib/luaproxy/)
+	add_subdirectory(lib/luaproxy)
 endif()
 
-# We use EXCLUDE_FROM_ALL so that only the explicit dependencies are used
-# (mbedTLS also has test and example programs in their CMakeLists.txt, we don't want those)
-include(lib/mbedtls.cmake EXCLUDE_FROM_ALL)
+# Selectively disable warnings in the level where the target is created:
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+	# Generated file has old-style casts, missing prototypes, and deprecated declarations
+	set_source_files_properties("${CMAKE_SOURCE_DIR}/src/Bindings/Bindings.cpp" PROPERTIES COMPILE_OPTIONS -Wno-everything)
 
-if(NOT MSVC AND "${CMAKE_SYSTEM_PROCESSOR}" MATCHES "arm")
-	# mbed TLS uses the frame pointer's register in inline assembly for its bignum implementation:
-	# https://tls.mbed.org/kb/development/arm-thumb-error-r7-cannot-be-used-in-asm-here
-	target_compile_options(mbedcrypto PRIVATE -fomit-frame-pointer)
+	# File failed to follow NHS guidelines on handwashing and has not maintained good hygiene
+	set_source_files_properties("${CMAKE_SOURCE_DIR}/src/IniFile.cpp" PROPERTIES COMPILE_OPTIONS -Wno-header-hygiene)
```

Comment:
Wrap in `if (UNITY_BUILDS)`?

Reply:
I think I'll fix the use namespace std, it's cleaner